platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/OutputWire.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+
+import com.swirlds.common.wiring.transformers.WireFilter;
+import com.swirlds.common.wiring.transformers.WireListSplitter;
+import com.swirlds.common.wiring.transformers.WireTransformer;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Describes the output of a task scheduler. Can be soldered to wire inputs or lambdas.
+ *
+ * @param <OUT> the output type of the object
+ */
+public abstract class OutputWire<OUT> {
+
+    private static final Logger logger = LogManager.getLogger(OutputWire.class);
+
+    private final WiringModel model;
+    private final String name;
+    private final List<Consumer<OUT>> forwardingDestinations = new ArrayList<>();
+
+    /**
+     * Constructor.
+     *
+     * @param model               the wiring model containing this output wire
+     * @param name                the name of the output wire
+     * @param registerWithModel   if true, then this output wire will be registered with the wiring model. Every time a
+     *                            new wire is created it should be registered with the wiring model. Secondary output
+     *                            wires should not be registered with the wiring model.
+     * @param insertionIsBlocking when data is inserted into this wire, will it block until capacity is available?
+     */
+    protected OutputWire(
+            @NonNull final WiringModel model,
+            @NonNull final String name,
+            final boolean registerWithModel,
+            final boolean insertionIsBlocking) {
+
+        this.model = Objects.requireNonNull(model);
+        this.name = Objects.requireNonNull(name);
+
+        if (registerWithModel) {
+            model.registerVertex(name, insertionIsBlocking);
```

Comment:
Conceptually, it does not make sense to call an output wire a vertex in a graph. The task scheduler + component combo is the vertex.